src/hooks/core/chat/useFetchChannelHook.ts

```diff
@@ -142,54 +166,81 @@ const useFetchChannelHook = () => {
+                    if (!isUserExists) {
+                      userMember.save(localDb).then(() => {
+                        return resolve(true);
+                      });
+                    } else {
                       return resolve(true);
-                    });
-                  } else {
-                    return resolve(true);
+                    }
                   }
-                }
-              );
-            });
-          }
-        });
+                );
+              });
+            }
+          });
+        }
 
         return null;
       });
 
       (channel?.messages || []).map((message) => {
-        queue.addJob({
-          label: `saveChat-${message?.id}`,
-          task: () => {
-            return new Promise((resolve) => {
-              if (
-                message?.type === 'deleted' ||
-                message?.message_type === 'notification-deleted' ||
-                message?.type === 'notification-deleted'
-              ) {
-                resolve(true);
-                return;
-              }
-              if (message?.type === 'system') {
-                message = getFirstMessage([message]);
-              }
-              const chat = ChatSchema.fromGetAllChannelAPI(channel?.id, message);
-              chat.saveIfNotExist(localDb).then(() => {
-                resolve(true);
-
-                refreshWithId('chat', channel?.id);
-              });
-            });
+        if (Platform.OS === 'android') {
+          if (
+            message?.type === 'deleted' ||
+            message?.message_type === 'notification-deleted' ||
+            message?.type === 'notification-deleted'
+          ) {
+            return;
+          }
+          if (message?.type === 'system') {
+            message = getFirstMessage([message]);
           }
-        });
+          const chat = ChatSchema.fromGetAllChannelAPI(channel?.id, message);
+          chat.saveIfNotExist(localDb).then(() => {
+            refreshWithId('chat', channel?.id);
+          });
+        } else {
+          queue.addJob({
+            label: `saveChat-${message?.id}`,
+            task: () => {
+              return new Promise((resolve) => {
+                if (
+                  message?.type === 'deleted' ||
+                  message?.message_type === 'notification-deleted' ||
+                  message?.type === 'notification-deleted'
+                ) {
+                  resolve(true);
+                  return;
+                }
+                if (message?.type === 'system') {
+                  message = getFirstMessage([message]);
+                }
+                const chat = ChatSchema.fromGetAllChannelAPI(channel?.id, message);
+                chat.saveIfNotExist(localDb).then(() => {
+                  resolve(true);
+
+                  refreshWithId('chat', channel?.id);
+                });
+              });
+            }
+          });
+        }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This block introduces a platform-specific conditional check for saving chat messages. On Android, it directly checks the message type and saves the chat message if appropriate. On other platforms, it adds this task to a queue for later processing. This seems like an optimization for Android, but it's important to ensure that this doesn't introduce any race conditions or inconsistencies in the application state.

```diff
+         if (Platform.OS === 'android') {
+           if (
+             message?.type === 'deleted' ||
+             message?.message_type === 'notification-deleted' ||
+             message?.type === 'notification-deleted'
+           ) {
+             return;
+           }
+           if (message?.type === 'system') {
+             message = getFirstMessage([message]);
+            }
+           const chat = ChatSchema.fromGetAllChannelAPI(channel?.id, message);
+           chat.saveIfNotExist(localDb).then(() => {
+             refreshWithId('chat', channel?.id);
+           });
+         } else {
+           queue.addJob({
+             label: `saveChat-${message?.id}`,
+             task: () => {
+               return new Promise((resolve) => {
+                 if (
+                   message?.type === 'deleted' ||
+                   message?.message_type === 'notification-deleted' ||
+                   message?.type === 'notification-deleted'
+                 ) {
+                   resolve(true);
+                   return;
+                 }
+                 if (message?.type === 'system') {
+                   message = getFirstMessage([message]);
+                 }
+                 const chat = ChatSchema.fromGetAllChannelAPI(channel?.id, message);
+                 chat.saveIfNotExist(localDb).then(() => {
+                   resolve(true);
+
+                   refreshWithId('chat', channel?.id);
+                 });
+               });
+             }
+           });
+         }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->